src/main/scala/strawman/collection/immutable/Seq.scala

```diff
@@ -2,16 +2,28 @@ package strawman
 package collection
 package immutable
 
-import strawman.collection.mutable.Builder
-import strawman.collection.IterableFactory
-
-import scala.Any
+import scala.{Any, Int}
 
 trait Seq[+A] extends Iterable[A]
                  with collection.Seq[A]
                  with SeqOps[A, Seq, Seq[A]]
 
-trait SeqOps[+A, +CC[A] <: Seq[A], +C] extends collection.SeqOps[A, CC, C]
+trait SeqOps[+A, +CC[X] <: Seq[X], +C] extends collection.SeqOps[A, CC, C] {
+
+  /** A copy of this $coll with one single replaced element.
+    *  @param  index  the position of the replacement
+    *  @param  elem   the replacing element
+    *  @tparam B        the element type of the returned $coll.
+    *  @return a new $coll which is a copy of this $coll with the element at position `index` replaced by `elem`.
+    *  @throws IndexOutOfBoundsException if `index` does not satisfy `0 <= index < length`.
+    *
+    *    @inheritdoc
+    *
+    *    @return a copy of this $coll with the element at position `index` replaced by `elem`.
+    */
+  def updated[B >: A](index: Int, elem: B): CC[B]
```

Comment:
I introduced the `updated` operation in *immutable* `Seq`. In the current collections, this operation takes an implicit `CanBuildFrom` that builds the resulting collection.

Here, I chose to leave it abstract (no default implementation) and implemented it in each concrete collection. For `List` I reused the same implementation as the current one (even though we could improve it by reusing the tail (the elements after `index`) instead of re-creating a new `List`), but for `Range`, `Vector` and `LazyList` I wrote custom implementations.